support-frontend/assets/pages/contributions-landing/components/PaymentMethodSelector.tsx

```diff
@@ -112,231 +104,277 @@ const mapDispatchToProps = {
+											)}
+										</div>
+									</>
+								),
+							)}
+
+						{paymentMethods.map((paymentMethod) => (
+							<Radio
+								id={`paymentMethodSelector-${paymentMethod}`}
+								name="paymentMethodSelector"
+								type="radio"
+								value={paymentMethod}
+								onChange={() => {
+									onPaymentMethodUpdate(paymentMethod);
+								}}
+								checked={props.paymentMethod === paymentMethod}
+								label={
+									<PaymentMethodLabel
+										label={getPaymentLabel(paymentMethod)}
+										logo={<PaymentMethodLogo paymentMethod={paymentMethod} />}
+										isChecked={props.paymentMethod === paymentMethod}
 									/>
-									<div
-										css={css`
-											font-size: small;
-											font-style: italic;
-											margin-left: 40px;
-											padding-bottom: 6px;
-											color: #767676;
-											padding-right: 40px;
-										`}
+								}
+							/>
+						))}
+
+						{contributionTypeIsRecurring(props.contributionType) &&
+							props.existingPaymentMethods &&
+							props.existingPaymentMethods.length > 0 &&
+							fullExistingPaymentMethods.length === 0 && (
+								<li className="form__radio-group-item">
+									...or{' '}
+									<a
+										className="reauthenticate-link"
+										href={getReauthenticateUrl()}
 									>
-										Used for your{' '}
-										{subscriptionsToExplainerList(
-											existingPaymentMethod.subscriptions.map(
-												subscriptionToExplainerPart,
-											),
-										)}
-									</div>
-								</>
-							),
-						)}
-					{paymentMethods.map((paymentMethod) => (
-						<Radio
-							id={`paymentMethodSelector-${paymentMethod}`}
-							name="paymentMethodSelector"
-							type="radio"
-							value={paymentMethod}
-							onChange={() => {
-								onPaymentMethodUpdate(paymentMethod, props);
-							}}
-							checked={props.paymentMethod === paymentMethod}
-							label={renderLabelAndLogo(paymentMethod)}
-							cssOverrides={radioCss}
-						/>
-					))}
-					{contributionTypeIsRecurring(props.contributionType) &&
-						props.existingPaymentMethods &&
-						props.existingPaymentMethods.length > 0 &&
-						fullExistingPaymentMethods.length === 0 && (
-							<li className="form__radio-group-item">
-								...or{' '}
-								<a
-									className="reauthenticate-link"
-									href={getReauthenticateUrl()}
-								>
-									re-enter your password
-								</a>{' '}
-								to use one of your existing payment methods.
-							</li>
-						)}
+										re-enter your password
+									</a>{' '}
+									to use one of your existing payment methods.
+								</li>
+							)}
+					</>
 				</RadioGroup>
 			) : (
-				noPaymentMethodsErrorMessage
+				<GeneralErrorMessage
+					classModifiers={['no-valid-payments']}
+					errorHeading="Payment methods are unavailable"
+					errorReason="all_payment_methods_unavailable"
+				/>
 			)}
 		</div>
 	);
 }
 
+// ----- Helper components ----- //
```

Comment:
We had a few functions (e.g `legend`) that were functions that returned `JSX`. That's exactly what a function component is! I've gone and refactored those functions into function components.